### ISSUE TITLE
Generate multiple identifier declarations with the same name (PatternPropertyJsonString)

### ISSUE THREAD

**Issue by moomiji** [corvus-dotnet/Corvus.JsonSchema#489]
Error CS0102

- Version: 4.0.7 Build: 876fe5b6941ca46f33214412c092b8b2f9699d4e

- test.json
```json
{
    "$schema": "https://json-schema.org/draft-07/schema",
    "type": "object",
    "patternProperties": {
        "^(?!.*?\\$schema).*$": {
            "type": "object",
            "patternProperties": {
                ".*_doc$|^doc$": {
                    "type": "string"
                },
                ".*_code$|^code$": {
                    "type": "string"
                }
            }
        }
    }
}
```

- Behavior in v3.1.1
```CSharp
        public static Regex PatternProperty1 => new(".*_doc$|^doc$", RegexOptions.Compiled, TimeSpan.FromSeconds(1));
        public static Regex PatternProperty2 => new(".*_code$|^code$", RegexOptions.Compiled, TimeSpan.FromSeconds(1));
```

**Comment by moomiji** [comment]
The `.Validate(ValidationContext.ValidContext, ValidationLevel.Verbose)` method also does not work, when only the `doc` property is kept.

- generate
```powershell
generatejsonschematypes .\test.json --rootNamespace JsonSchemaSample.Test --outputRootTypeName Test --outputPath .\test
```

- test code
```CSharp
using Corvus.Json;
using JsonSchemaSample.Test;

var test = """
{
    "a_doc": "aaa",
    "b_doc": "bbb"
}
""";

var document = Test.SchemaEntity.Parse(test);
var result = document.Validate(ValidationContext.ValidContext, ValidationLevel.Verbose);
```

- error
```
Unhandled exception. System.InvalidOperationException: This operation does not apply to an empty instance.
   at System.Collections.Immutable.ImmutableStack`1.Pop()
   at Corvus.Json.ValidationContext.PopLocation()
   at JsonSchemaSample.Test.Test.SchemaEntity.Validate(ValidationContext& validationContext, ValidationLevel level) in C:\Users\73734\source\JsonSchemaSample\test\Test.SchemaEntity.Validate.cs:line 62
   at Program.<Main>$(String[] args) in C:\Users\73734\source\JsonSchemaSample\Program.cs:line 300
```

**Comment by mwadams** [comment]
Thanks for the repro. We will get that fixed ASAP.

**Comment by mwadams** [comment]
@moomiji - Can you provide an illustration of the second problem?

I cannot repro with this schema:

```
	Given a schema file
		"""
		{
			"$schema": "https://json-schema.org/draft-07/schema",
			"type": "object",
			"patternProperties": {
				".*_doc$|^doc$": {
					"type": "string"
				}
			}
		}
		"""
```

and this doc

```
{ "a_doc": "aaa", "b_doc": "bbb" }
```

**Comment by moomiji** [comment]
[Repro489.zip](https://github.com/user-attachments/files/17798373/Repro489.zip)

I created a new project to reproduce the second problem by following these steps.

```powershell
dotnet new console -o Repro489 -f net8.0
cd Repro489
dotnet add package Corvus.Json.ExtendedTypes --version 4.0.7

dotnet new tool-manifest
dotnet tool install Corvus.Json.JsonSchema.TypeGeneratorTool --version 4.0.7

code test.json
generatejsonschematypes .\test.json --rootNamespace Repro489.Test --outputPath .\test

code Program.cs
dotnet run
```

**Comment by mwadams** [comment]
@moomiji - Fixed in 4.0.9
